src/tests/ftest/util/bytes_utils.py

```diff
@@ -0,0 +1,576 @@
+#!/usr/bin/python
+"""
+(C) Copyright 2020 Intel Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+The Government's rights to use, modify, reproduce, release, perform, display,
+or disclose this software are subject to the terms of the Apache License as
+provided in Contract No. B609815.
+Any reproduction of computer software, computer software documentation, or
+portions thereof marked with this legend must also reproduce the markings.
+"""
+from logging import getLogger
+
+
+class ByteUnit(object):
+    # pylint: disable=too-few-public-methods
+    """Defines the unit for a Byte object."""
+
+    VALUES = ("", "K", "M", "G", "T", "P", "E", "Z", "Y")
+
+    def __init__(self, value=None):
+        """Initialize a ByteUnit object with a unit.
+
+        Args:
+            value (str, optional): unit value. Defaults to None.
+        """
+        self._value = None
+        if value is None:
+            self._value = self.VALUES[0]
+        elif value in self.VALUES:
```

Comment:
The support for a lower case string is handled in the next `elif isinstance(value, str) and value.upper()[0] in self.VALUES` statement.  We cannot simply issue a `value.upper()` here because there is no guarantee that the `value` object has an `upper()` attribute, i.e. `value` could be an `int`.

Reply:
I remember why I did the checks this way.  It is due to this exception:
```
>>> "".upper()[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: string index out of range
```
Since `""` is a valid `self.VALUES`, I need to check first to see if `""` has been provided before converting to a string and checking if the first uppercase character of the string is in `self.VALUES`